### ISSUE TITLE
Enable CSRF, disable remote CLI per default in the jenkins chart

### ISSUE THREAD

**Issue by sboardwell** [jenkins-x/jenkins-x-platform#4615]
As discussed in [the slack channel](https://kubernetes.slack.com/archives/C9LTHT2BB/p1547030663197700) with @jstrachan and @jglick the Jenkins instance should be security hardened per default. This can be done through 
* the use of an init script in the current jenkins chart [here](https://github.com/jenkins-x/charts/blob/jenkins/stable/jenkins/templates/config.yaml#L278)
* later through the use of the specific config parameters in the chart [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/config.yaml#L149) - this is dependent on the current fork of the charts being rebased.
* eventually @jglick mentioned it would be better to use the `configuration-as-code` plugin instead of templating the `config.xml` - I would see this being carried upstream in the official helm repo though.

**Comment by jglick** [comment]
You should not need an init script; just add

```xml
<crumbIssuer class="hudson.security.csrf.DefaultCrumbIssuer"/>
```

to `config.xml`. Similarly, easy to backport `jenkins.CLI.xml`:

```xml
<jenkins.CLI>
    <enabled>false</enabled>
</jenkins.CLI>
```

Do not bother even making these things configurable IMO. Anyway https://github.com/jenkinsci/jenkins/pull/3838 would remove the need for the latter.

> use the `configuration-as-code` plugin

[Reference](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/README.md)

**Comment by sboardwell** [comment]
Good catch. Didn't think of that since I only used the init script whilst building a docker image. Extending the `config.xml` is much easier (and 👍 for the "...not bother even making these things configurable..." ).

**Comment by jglick** [comment]
Also see: https://github.com/jenkinsci/jep/pull/238

**Comment by jglick** [comment]
As of https://github.com/jenkinsci/jenkins/pull/4509 the CSRF issuer would always be on.

Anyway my reading of https://jenkins-x.io/blog/2020/03/11/tekton/ is that this will be closed.
